QModManager/API/QModServices.cs

```diff
@@ -116,5 +117,19 @@ public IQMod GetMod(string id)
         {
             return FindModById(id);
         }
+
+        /// <summary>
+        /// Adds a critical message to the main menu.
+        /// Message will stay in the main menu and on the loading screen.
+        /// </summary>
+        /// <param name="msg">The message to add.</param>
+        /// <param name="size">The size of the text.</param>
+        /// <param name="color">The color of the text.</param>
+        /// <param name="autoformat">Whether or not to apply formatting tags to the message, or show it as it is.</param>
+        public void AddCriticalMessage(string msg, int size = MainMenuMessages.defaultSize, string color = MainMenuMessages.defaultColor, bool autoformat = true)
+        {
+            var callingMod = GetMod(ReflectionHelper.CallingAssemblyByStackTrace());
```

Comment:
Will this 100% every time be the correct assembly?

Reply:
You mean will it be the correct mod? That `CallingAssemblyByStackTrace()` ignores QMM's assembly and system assembly, so it should be correct, I think.